addons/base_vat/models/account_fiscal_position.py

```diff
@@ -13,12 +13,27 @@ def _validate_foreign_vat(self):
             if not record.foreign_vat:
                 continue
 
-            checked_country_code = self.env['res.partner']._run_vat_test(record.foreign_vat, record.country_id)
+            checked_country_code = False
 
-            if checked_country_code and checked_country_code != record.country_id.code.lower():
-                raise ValidationError(_("The country detected for this foreign VAT number does not match the one set on this fiscal position."))
+            if record.country_id:
+                checked_country_code = self.env['res.partner']._run_vat_test(record.foreign_vat, record.country_id)
+                if checked_country_code and checked_country_code != record.country_id.code.lower():
+                    if not record.country_group_id:
+                        raise ValidationError(_("The country detected for this foreign VAT number does not match the country on this fiscal position."))
+                if not checked_country_code:
+                    fp_label = _("fiscal position [%s]", record.name)
+                    error_message = self.env['res.partner']._build_vat_error_message(record.country_id.code.lower(), record.foreign_vat, fp_label)
+                    raise ValidationError(error_message)
 
-            if not checked_country_code:
-                fp_label = _("fiscal position [%s]", record.name)
-                error_message = self.env['res.partner']._build_vat_error_message(record.country_id.code.lower(), record.foreign_vat, fp_label)
-                raise ValidationError(error_message)
+            elif record.country_group_id:
+                foreign_vat_country_id = False
+                for country_id in record.country_group_id.country_ids:
+                    foreign_vat_country_id = country_id if record.foreign_vat[:2] == country_id.code.upper() else foreign_vat_country_id
+                if not foreign_vat_country_id:
+                    raise ValidationError(_("The country detected for this foreign VAT number does not match any of the countries composing the country group set on this fiscal position."))
+                record.country_id = foreign_vat_country_id
+                checked_country_code = self.env['res.partner']._run_vat_test(record.foreign_vat, foreign_vat_country_id)
+                if not checked_country_code:
+                    fp_label = _("fiscal position [%s]", record.name)
+                    error_message = self.env['res.partner']._build_vat_error_message(foreign_vat_country_id.code.lower(), record.foreign_vat, fp_label)
+                    raise ValidationError(error_message)
```

Comment:
I think that this duplicated code could be put in a function to have a better readability